include/quic/parser.hpp

```diff
@@ -0,0 +1,380 @@
+                    log::debug(bp_cat, "Successfully matched response to sent request!");
+                    sent_reqs.erase(r.first);
+                    return true;
+                }
+            }
+
+            return false;
+        }
+
+        void handle_input(message msg)
+        {
+            log::debug(bp_cat, "{} called", __PRETTY_FUNCTION__);
+
+            if (msg.req_type == "R" || msg.req_type == "E")
+            {
+                if (auto b = match(msg.req_id); not b)
+                {
+                    log::warning(bp_cat, "Error: could not match orphaned response!");
+                    return;
+                }
+            }
+
+            recv_callback(*this, std::move(msg));
+        }
+
+        void process_incoming(std::string_view req)
+        {
+            log::debug(bp_cat, "{} called", __PRETTY_FUNCTION__);
+
+            while (not req.empty())
+            {
+                if (current_len == 0)
+                {
+                    int consumed = 0;
+
+                    if (not size_buf.empty())
+                    {
+                        int prev_len = size_buf.size();
+                        size_buf += req.substr(0, 15);
+
+                        consumed = parse_length(size_buf);
+
+                        if (consumed == 0)
+                            return;
+
+                        size_buf.clear();
+                        req.remove_prefix(consumed - prev_len);
+                    }
+                    else
+                    {
+                        consumed = parse_length(req);
+                        if (consumed == 0)
+                        {
+                            size_buf += req;
+                            return;
+                        }
+
+                        req.remove_prefix(consumed);
+                    }
+
+                    if (req.size() >= current_len)
+                    {
+                        buf += req.substr(0, current_len);
+                        handle_input(message{std::move(buf)});
+                        req.remove_prefix(current_len);
+
+                        current_len = 0;
+                        continue;
+                    }
+
+                    buf.reserve(current_len);
+                    buf += req;
+                    return;
+                }
+
+                auto r_size = req.size() + buf.size();
+
+                if (r_size >= current_len)
+                {
+                    buf += req.substr(0, r_size);
+                    req.remove_prefix(r_size);
+                    handle_input(message{std::move(buf)});
+                    current_len = 0;
+                    continue;
+                }
+
+                buf += req;
+                return;
+            }
+        }
+
+        std::shared_ptr<sent_request> make_request(std::string endpoint, std::string body)
+        {
+            oxenc::bt_list_producer btlp;
+            std::string rid = std::to_string(++next_rid);
```

Comment:
Probably easier just to put an int in the field, i.e.:
```suggestion
            auto rid = ++next_rid;
```
here and when we parse it, `consume_integer<int64_t>()` instead of consuming/parsing a string (in the message constructor, around line 36).

Reply:
Nice i didn't realize there were templated int parsing/consuming arguments in oxenc, thank you